packages/foundry/contracts/FYToken.sol

```diff
@@ -86,6 +86,36 @@ contract FYToken is IFYToken, IERC3156FlashLender, AccessControl, ERC20Permit, C
         emit FlashFeeFactorSet(flashFeeFactor_);
     }
 
+    ///@dev Converts the amount of the principal to the underlying
+    function convertToUnderlying(uint256 principalAmount) external override returns (uint256 underlyingAmount) {
+        return _convertToUnderlying(principalAmount);
+    }
+
+    ///@dev Converts the amount of the principal to the underlying
+    ///Before maturity, returns amount as if at maturity.
+    function _convertToUnderlying(uint256 principalAmount) internal returns (uint256 underlyingAmount) {
+        if (chiAtMaturity == CHI_NOT_SET) {
+            return principalAmount;
+        } else {
+            return _accrual();
```

Comment:
Wouldn't this be `underlyingAmount.wmul(_accrual())`?

Does the EIP say anything about rounding?

Could you please code the unit tests that catch this bug first, and then fix it?

Reply:
It would, yes.